crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -676,11 +688,23 @@ impl Object {
         address: SuiAddress,
         native: NativeObject,
         checkpoint_viewed_at: u64,
+        root_version: Option<u64>,
     ) -> Object {
+        let root_version = root_version.or_else(|| Self::infer_root_version(&native));
         Object {
             address,
             kind: ObjectKind::NotIndexed(native),
             checkpoint_viewed_at,
+            root_version,
+        }
+    }
+
+    fn infer_root_version(native: &NativeObject) -> Option<u64> {
+        match native.as_inner().owner {
+            NativeOwner::AddressOwner(_) | NativeOwner::Shared { .. } => {
+                Some(native.as_inner().version().into())
+            }
+            _ => None,
```

Comment:
nit: i think this can just be a standalone function so we don't need the prefix `Self::`